src/shared/codelens/pythonCodeLensProvider.ts

```diff
@@ -59,75 +70,210 @@ const getLambdaHandlerCandidates = async ({ uri }: { uri: vscode.Uri }): Promise
+
+const makeDebugConfig = ({debugPort, samProjectCodeRoot}: {
+    debugPort?: number,
+    samProjectCodeRoot: string,
+}): PythonDebugConfiguration => {
+    return {
+        type: PYTHON_LANGUAGE,
+        request: 'attach',
+        name: 'SamLocalDebug',
+        host: 'localhost',
+        port: debugPort!,
+        pathMappings: [
+            {
+                // tslint:disable-next-line:no-invalid-template-strings
+                localRoot: fixFilePathCapitalization(samProjectCodeRoot),
+                remoteRoot: '/var/task'
+            }
+        ],
+    }
+}
+
+export async function initialize({
     configuration,
     toolkitOutputChannel,
     processInvoker = new DefaultSamCliProcessInvoker(),
     taskInvoker = new DefaultSamCliTaskInvoker()
-}: CodeLensProviderParams): void {
+}: CodeLensProviderParams): Promise<void> {
     const logger = getLogger()
 
-    const runtime = 'python3.7' // TODO: Remove hard coded value
+    const runtime = 'python3.6' // TODO: Remove hard coded value
 
-    const invokeLambda = async (args: LambdaLocalInvokeArguments) => {
+    const invokeLambda = async (args: LambdaLocalInvokeParams) => {
         const samProjectCodeRoot = await getSamProjectDirPathForFile(args.document.uri.fsPath)
-        logger.debug(`Project root: '${samProjectCodeRoot}'`)
+        const baseBuildDir = await makeBuildDir()
+
         let debugPort: number | undefined
 
+        let handlerName: string
+        let manifestPath: string | undefined
         if (args.isDebug) {
             debugPort = await getDebugPort()
+            const {debugHandlerName} = await makeLambdaDebugFile({
+                handlerName: args.handlerName,
+                debugPort: debugPort,
+                outputDir: samProjectCodeRoot,
+            })
+            handlerName = debugHandlerName
+            manifestPath = await makePythonDebugManifest({
+                samProjectCodeRoot,
+                outputDir: baseBuildDir
+            })
+        } else {
+            handlerName = args.handlerName
         }
-        // TODO: Figure out Python specific params and create PythonDebugConfiguration if needed
-        const debugConfig: DebugConfiguration = {
-            type: PYTHON_LANGUAGE,
-            request: 'attach',
-            name: 'SamLocalDebug',
-            preLaunchTask: undefined,
-            address: 'localhost',
-            port: debugPort!,
-            localRoot: samProjectCodeRoot,
-            remoteRoot: '/var/task',
-            skipFiles: []
-        }
+        const inputTemplatePath = await makeInputTemplate({
+            baseBuildDir,
+            codeDir: samProjectCodeRoot,
+            documentUri: args.document.uri,
+            handlerName,
+            runtime,
+            workspaceUri: args.workspaceFolder.uri
+        })
+        logger.info(`pythonCodeLensProvider.initialize: ${
+            JSON.stringify({samProjectCodeRoot, inputTemplatePath, handlerName, manifestPath}, undefined, 2)
```

Comment:
In other places, I see log messages `stringified` without specifying the tab-width. We should be consistent across our codebase with log tab widths.

Reply:
Perhaps.. but util this debugging is done I surely don't want to look at a bunch of crazy long lines of JSON to try and visually parse the tokens as would be the case if I simply had `JSON.stringify(obj)`. I suppose it depends if we want to optimize logs for `grep` or for reading. In this case, for me they are and have been used for reading within VS Code. My intension was to change these before release... at least by changing them from `info` to `debug`. 

Longer term, consider making logger accept objects in its argument list.

1. Objects could be pretty printed to channel (by "specifying the tab-width")
2. Object written to file logs could use basic `JSON.stringify` _if_ we determine that helps with grep or other tools used to search logs. As mentioned in the log discussion, we may consider logging to a JSONl formatted file to more readily support parsing, ingesting, processing...

For now the only use case I have is reading so if that's the norm for the team, perhaps we should favor readability. Feel free to create an issue if you feel it's worth updating elsewhere but it seems counter-productive to avoid making any logs readable until we have made them all readable. 

Added a `TODO` to clean up log statements before release.

Reply:
I didn't mean that we should increase the tab width, just that we should use the same tab-width (short or long) throughout this change.

Reply:
It's not about tab-width it's pretty print or one long line of un-pretty JSON.

Reply:
It appears @mpiroc misunderstood what I was doing. Resolving conversation.